@ararslan I build this PR from cloning it into a new dir, but got following error
```
LoadError("sysimg.jl", 362, LoadError("sparse/sparse.jl", 41, LoadError("sparse/umfpack.jl", 61, ErrorException("error compiling anonymous: could not load library \"libsuitesparse_wrapper\"\n/lib/libgcc_s.so.1: version GCC_4.6.0 required by /usr/local/lib/gcc5/libgfortran.so.3 not found"))))
*** This error is usually fixed by running `make clean`. If the error persists, try `make cleanall`. ***
gmake[1]: *** [Makefile:232: /usr/home/iblis/tmp/jl2/usr/lib/julia/sys.o] Error 1
gmake: *** [Makefile:109: julia-sysimg-release] Error 2
```

(this error msg is same as `-B` + remove `fixup.sh`)

I'm on an old -CURRENT
```
└─[iblis@abeing]% uname -a
FreeBSD abeing 12.0-CURRENT FreeBSD 12.0-CURRENT #4 r309367: Mon Dec 12 11:10:09 CST 2016     root@abeing:/usr/obj/usr/src/sys/GENERIC  amd64
```

--- a/Make.inc
+++ b/Make.inc
@@ -520,6 +520,36 @@ ifeq (exists, $(shell [ -e $(BUILDROOT)/Make.user ] && echo exists ))
 include $(BUILDROOT)/Make.user
 endif
 
+# A bit of a kludge to work around libraries linking to FreeBSD's outdated system libgcc_s
+# Instead, let's link to the libgcc_s corresponding to the installation of gfortran
+ifeq ($(OS),FreeBSD)
+ifneq (,$(findstring gfortran,$(FC)))
+
+# First let's figure out what version of GCC we're dealing with
+_GCCMAJOR := $(shell $(FC) -dumpversion | cut -d'.' -f1)
+_GCCMINOR := $(shell $(FC) -dumpversion | cut -d'.' -f2)
+
+# The ports system uses major and minor for GCC < 5 (e.g. gcc49 for GCC 4.9), otherwise major only
+ifeq ($(_GCCMAJOR),4)
+  _GCCVER := $(_GCCMAJOR)$(_GCCMINOR)
+else
+  _GCCVER := $(_GCCMAJOR)
+endif
+
+# Allow the user to specify this in Make.user
+GCCPATH ?= $(LOCALBASE)/lib/gcc$(_GCCVER)
+
+LDFLAGS += -L$(build_libdir) -L$(GCCPATH) -Wl,-rpath,$(build_libdir) -Wl,-rpath,$(GCCPATH)
+
+# This ensures we get the right RPATH even if we're missing FFLAGS somewhere
+FC += -Wl,-rpath=$(GCCPATH)
+
+# Build our own libc++ and libc++abi because otherwise /usr/lib/libc++.so and /lib/libcxxrt.so will
+# be linked in when building LLVM, and those link to /lib/libgcc_s.so
+BUILD_CUSTOM_LIBCXX ?= 1
+endif # gfortran
+endif # FreeBSD
+
 ifneq ($(CC_BASE)$(CXX_BASE),$(shell echo $(CC) | cut -d' ' -f1)$(shell echo $(CXX) | cut -d' ' -f1))
     $(error Forgot override directive on CC or CXX in Make.user? Cowardly refusing to build)
 endif

--- a/deps/libgit2.mk
+++ b/deps/libgit2.mk
@@ -37,14 +37,7 @@ LIBGIT2_OPTS += -DCURL_INCLUDE_DIRS=$(build_includedir) -DCURL_LIBRARIES="-L$(bu
 endif
 
 ifeq ($(OS),Linux)
-LIBGIT2_OPTS += -DUSE_OPENSSL=OFF -DUSE_MBEDTLS=ON -DCMAKE_INSTALL_RPATH="\$$ORIGIN"
-endif
-
-ifeq ($(OS),FreeBSD)
-# We're getting this from CMAKE_COMMON when GCCPATH is nonempty
-ifeq ($(GCCPATH),)
-LIBGIT2_OPTS += -DCMAKE_INSTALL_RPATH="\$$ORIGIN"
-endif
+LIBGIT2_OPTS += -DUSE_OPENSSL=OFF -DUSE_MBEDTLS=ON
 endif
 
 # We need to bundle ca certs on linux now that we're using libgit2 with ssl

--- a/deps/libgit2.mk
+++ b/deps/libgit2.mk
@@ -41,7 +41,10 @@ LIBGIT2_OPTS += -DUSE_OPENSSL=OFF -DUSE_MBEDTLS=ON -DCMAKE_INSTALL_RPATH="\$$ORI
 endif
 
 ifeq ($(OS),FreeBSD)
+# We're getting this from CMAKE_COMMON when GCCPATH is nonempty
+ifeq ($(GCCPATH),)
 LIBGIT2_OPTS += -DCMAKE_INSTALL_RPATH="\$$ORIGIN"
+endif
 endif
 
 # We need to bundle ca certs on linux now that we're using libgit2 with ssl

--- a/deps/libssh2.mk
+++ b/deps/libssh2.mk
@@ -20,14 +20,6 @@ else
 LIBSSH2_OPTS += -DCRYPTO_BACKEND=mbedTLS -DENABLE_ZLIB_COMPRESSION=OFF
 endif
 
-ifeq ($(OS),Linux)
-LIBSSH2_OPTS += -DCMAKE_INSTALL_RPATH="\$$ORIGIN"
-endif
-
-ifeq ($(OS),FreeBSD)
-LIBSSH2_OPTS += -DCMAKE_INSTALL_RPATH="\$$ORIGIN"
-endif
-
 $(SRCDIR)/srccache/$(LIBSSH2_SRC_DIR)/libssh2-encryptedpem.patch-applied: $(SRCDIR)/srccache/$(LIBSSH2_SRC_DIR)/source-extracted
 	cd $(SRCDIR)/srccache/$(LIBSSH2_SRC_DIR) && patch -p1 -f < $(SRCDIR)/patches/libssh2-encryptedpem.patch
 	echo 1 > $@

--- a/deps/llvm.mk
+++ b/deps/llvm.mk
@@ -138,7 +138,6 @@ endif # BUILD_LLDB
 # Part of the FreeBSD libgcc_s kludge
 ifeq ($(OS),FreeBSD)
 ifneq ($(GCCPATH),)
-LLVM_CMAKE += -DCMAKE_INSTALL_RPATH="\$$ORIGIN:$(GCCPATH)"
 LLVM_LDFLAGS += -Wl,-rpath,'\$$ORIGIN',-rpath,$(GCCPATH)
 endif
 endif

--- a/deps/llvm.mk
+++ b/deps/llvm.mk
@@ -135,6 +135,14 @@ LLVM_CMAKE += -DLLDB_DISABLE_PYTHON=ON
 endif # LLDB_DISABLE_PYTHON
 endif # BUILD_LLDB
 
+# Part of the FreeBSD libgcc_s kludge
+ifeq ($(OS),FreeBSD)
+ifneq ($(GCCPATH),)
+LLVM_CMAKE += -DCMAKE_INSTALL_RPATH="\$$ORIGIN:$(GCCPATH)"
+LLVM_LDFLAGS += -Wl,-rpath,'\$$ORIGIN',-rpath,$(GCCPATH)
+endif
+endif
+
 ifneq (,$(filter $(ARCH), powerpc64le ppc64le))
 LLVM_CXXFLAGS += -mminimal-toc
 endif

--- a/deps/mbedtls.mk
+++ b/deps/mbedtls.mk
@@ -20,7 +20,10 @@ MBEDTLS_OPTS += -DCMAKE_INSTALL_RPATH="\$$ORIGIN"
 endif
 
 ifeq ($(OS),FreeBSD)
+# We're getting this from CMAKE_COMMON when GCCPATH is nonempty
+ifeq ($(GCCPATH),)
 MBEDTLS_OPTS += -DCMAKE_INSTALL_RPATH="\$$ORIGIN"
+endif
 endif
 
 $(SRCDIR)/srccache/$(MBEDTLS_SRC).tgz: | $(SRCDIR)/srccache

--- a/deps/mbedtls.mk
+++ b/deps/mbedtls.mk
@@ -15,17 +15,6 @@ ifeq ($(BUILD_OS),WINNT)
 MBEDTLS_OPTS += -G"MSYS Makefiles"
 endif
 
-ifeq ($(OS),Linux)
-MBEDTLS_OPTS += -DCMAKE_INSTALL_RPATH="\$$ORIGIN"
-endif
-
-ifeq ($(OS),FreeBSD)
-# We're getting this from CMAKE_COMMON when GCCPATH is nonempty
-ifeq ($(GCCPATH),)
-MBEDTLS_OPTS += -DCMAKE_INSTALL_RPATH="\$$ORIGIN"
-endif
-endif
-
 $(SRCDIR)/srccache/$(MBEDTLS_SRC).tgz: | $(SRCDIR)/srccache
 	$(JLDOWNLOAD) $@ $(MBEDTLS_URL)
 

--- a/deps/tools/common.mk
+++ b/deps/tools/common.mk
@@ -37,12 +37,16 @@ CMAKE_COMMON += -DCMAKE_RC_COMPILER="$$(which $(CROSS_COMPILE)windres)"
 endif
 endif
 
+ifneq (,$(findstring $(OS),Linux FreeBSD))
+INSTALL_RPATH := "\$$ORIGIN"
 # Part of the FreeBSD libgcc_s kludge
 ifeq ($(OS),FreeBSD)
 ifneq ($(GCCPATH),)
-CMAKE_COMMON += -DCMAKE_INSTALL_RPATH="\$$ORIGIN:$(GCCPATH)"
+INSTALL_RPATH := "\$$ORIGIN:$(GCCPATH)"
 endif
 endif
+CMAKE_COMMON += -DCMAKE_INSTALL_RPATH=$(INSTALL_RPATH)
+endif # Linux or FreeBSD
 
 # For now this is LLVM specific, but I expect it won't be in the future
 ifeq ($(LLVM_USE_CMAKE),1)

--- a/deps/tools/common.mk
+++ b/deps/tools/common.mk
@@ -37,6 +37,13 @@ CMAKE_COMMON += -DCMAKE_RC_COMPILER="$$(which $(CROSS_COMPILE)windres)"
 endif
 endif
 
+# Part of the FreeBSD libgcc_s kludge
+ifeq ($(OS),FreeBSD)
+ifneq ($(GCCPATH),)
+CMAKE_COMMON += -DCMAKE_INSTALL_RPATH="\$$ORIGIN:$(GCCPATH)"
+endif
+endif
+
 # For now this is LLVM specific, but I expect it won't be in the future
 ifeq ($(LLVM_USE_CMAKE),1)
 ifeq ($(CMAKE_GENERATOR),Ninja)

